js/home_redirect_lang.language_switcher.js

```diff
@@ -16,6 +16,12 @@
   Drupal.behaviors.homeRedirectLangSwitcher = {
     attach: function (context) {
       let links = document.querySelectorAll('.language-switcher-language-url .language-link');
+
+      // Don't process when standard language switcher not found.
+      if (links.length === 0) {
```

Comment:
Ok found this https://github.com/antistatique/shop.swisstopo.ch/pull/336#discussion_r963053353

So the issue is not in the js of the module itself but in the doc snippet we propose

Reply:
better now ?

Reply:
https://ryandejaegher.com/avoid-errors-document-queryselector-document-queryselectorall/

> When querySelector doesn’t find a matching selector, the value of the variable will be null.

> querySelectorAll behaves differently. If it doesn’t find any elements it will still be defined as a NodeList. Which means if we check whether our element is truthy or falsy, it will return true.

you're right